settings.gradle.kts

```diff
@@ -15,34 +15,45 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
+import com.gradle.enterprise.gradleplugin.internal.extension.BuildScanExtensionWithHiddenFeatures
 
 plugins {
-  id("com.gradle.enterprise") version "3.4.1" apply false
+  id("com.gradle.enterprise") version "3.13.2"
+  id("com.gradle.common-custom-user-data-gradle-plugin") version "1.10"
 }
 
 
-// Plugins which require online access should not be enabled when running in offline mode.
-if (!gradle.startParameter.isOffline) {
-  apply(plugin = "com.gradle.enterprise")
-}
-
 // JENKINS_HOME and BUILD_ID set automatically during Jenkins execution
 val isJenkinsBuild = arrayOf("JENKINS_HOME", "BUILD_ID").all { System.getenv(it) != null }
 // GITHUB_REPOSITORY and GITHUB_RUN_ID set automatically during Github Actions run
 val isGithubActionsBuild = arrayOf("GITHUB_REPOSITORY", "GITHUB_RUN_ID").all { System.getenv(it) != null }
-if (isJenkinsBuild || isGithubActionsBuild) {
-  gradleEnterprise {
-    buildScan {
-      // Build Scan enabled and TOS accepted for Jenkins lab build. This does not apply to builds on
-      // non-Jenkins machines. Developers need to separately enable and accept TOS to use build scans.
-      termsOfServiceUrl = "https://gradle.com/terms-of-service"
-      termsOfServiceAgree = "yes"
-      publishAlways()
+val isCi = isJenkinsBuild || isGithubActionsBuild
+
+gradleEnterprise {
+  server = "https://ge.apache.org"
+  allowUntrustedServer = false
+
+  buildScan {
+    capture { isTaskInputFiles = true }
+    isUploadInBackground = !isCi
+    publishAlways()
+    this as BuildScanExtensionWithHiddenFeatures
+    publishIfAuthenticated()
+    obfuscation {
+      ipAddresses { addresses -> addresses.map { "0.0.0.0" } }
     }
   }
 }
 
+buildCache {
+  local {
+    isEnabled = !isCi
+  }
+  remote(gradleEnterprise.buildCache) {
+    isEnabled = false
```

Comment:
I see this is deliberately disabled. What is the reasoning behind this?

Reply:
Update: I see

> Very soon, we will follow up this PR with one that connects the Apache Beam build to a remote build cache node solely for the Apache Beam project to further help speed up builds with build caching.

So this is intended. However can we enable it for now to mitigate the worker pressures?

Reply:
The issue is actually line 50 which changes the behavior. There was never a remote build cache enabled for this build. I've submitted this PR to fix the issue: https://github.com/apache/beam/pull/26784